src/Containers/DenseAxisArray.jl

```diff
@@ -60,9 +60,21 @@ end
 
 function Base.getindex(
     x::_AxisLookup{Dict{K,Int}},
-    key::K,
-) where {K<:AbstractVector}
-    return x.data[key]
+    key::T,
+) where {T<:AbstractVector,K>:T}
+    # In this method, we can't tell if `keys` is an actual key, or a vector of
+    # keys. One common cause happens when `K` is `Any`.
+    if haskey(x.data, key)
+        if all(haskey(x.data, k) for k in key)
+            error(
+                "ambiguous use of getindex with key $key. We cannot tell if " *
+                "you meant to return the single element corresponding to the " *
+                "key, or a slice for each element in the key.",
```

Comment:
So I've opted just to error in the strict case that you index an ambiguous case. This is kinda similar to method ambiguities. You can define them, but you only error if there is a confusion at runtime. You could imagine someone defining an ambiguous set an then converting to `Array(x)` and being fine.